app/src/pages/FormPage.js

```diff
@@ -190,11 +178,79 @@ const FormPage = ({ navigation, route }) => {
     } catch (err) {
       console.error(err);
       if (Platform.OS === 'android') {
-        ToastAndroid.show(trans.errorSubmitted, ToastAndroid.LONG);
+        ToastAndroid.show(trans.errorSaveDatapoint, ToastAndroid.LONG);
       }
     }
   };
 
+  const handleOnSubmitForm = async (values) => {
+    setSubmitting({
+      text: trans.loadingText,
+      status: true,
+    });
+    const answers = {};
+    const photos = [];
+    formJSON.question_group
+      .flatMap((qg) => qg.question)
+      .forEach((q) => {
+        let val = values.answers?.[q.id];
+        if (!val && val !== 0) {
+          return;
+        }
+        if (q.type === 'cascade') {
+          val = val.slice(-1)[0];
+        }
+        if (q.type === 'number') {
+          val = parseFloat(val);
+        }
+        if (q.type === 'photo' && val) {
+          photos.push({ id: q.id, value: val });
+        }
+        answers[q.id] = val;
+      });
+    if (photos.length) {
+      const uploads = photos.map((p) => {
+        const fileType = p.value.split('.').slice(-1)[0];
+        const formData = new FormData();
+        formData.append('file', {
+          uri: p.value,
+          name: `photo_${currentFormId}_${p.id}.${fileType}`,
+          type: `image/${fileType}`,
+        });
+        return api.post('/images', formData, {
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'multipart/form-data',
+          },
+        });
+      });
+
+      axios
+        .all(uploads)
```

Comment:
How this is working? the image uploads requires a token.
Please check: http://localhost:3000/api/doc/#/Mobile%20Device%20Form/v1_device_images_create